candid/evm_rpc.did

```diff
@@ -0,0 +1,163 @@
+type Auth = variant { Rpc; RegisterProvider; FreeRpc; ManageService };
+type Block = record { base_fee_per_gas : nat; number : nat };
+type BlockSpec = variant { Tag : BlockTag; Number : nat };
+type BlockTag = variant {
+  Earliest;
+  Safe;
+  Finalized;
+  Latest;
+  Number : nat64;
+  Pending;
+};
+type CandidRpcSource = variant {
+  EthSepolia : opt SepoliaProvider;
+  EthMainnet : opt SepoliaProvider;
+};
+type DataFormatError = variant { InvalidHex : text };
+type FeeHistory = record {
+  reward : vec vec nat;
+  base_fee_per_gas : vec nat;
+  oldest_block : nat;
+};
+type FeeHistoryArgs = record {
+  block_count : nat;
+  newest_block : BlockSpec;
+  reward_percentiles : opt vec nat8;
+};
+type GetLogsArgs = record {
+  addresses : vec text;
+  topics : opt vec text;
+  to_block : opt BlockSpec;
+  from_block : opt BlockSpec;
+};
+type GetTransactionCountArgs = record { address : text; block : BlockSpec };
+type HttpOutcallError = variant {
+  IcError : record { code : RejectionCode; message : text };
+  InvalidHttpJsonRpcResponse : record {
+    status : nat16;
+    body : text;
+    parsing_error : opt text;
+  };
+};
+type JsonRpcError = record { code : int64; message : text };
+type LogEntry = record {
+  transaction_hash : opt vec nat8;
+  block_hash : opt vec nat8;
+  log_index : opt nat;
+  data : vec nat8;
+  transaction_index : opt nat;
+  block_number : opt nat;
+  topics : vec vec nat8;
+  address : vec nat8;
+  removed : bool;
+};
+type Message = variant { Data : vec nat8; Hash : vec nat8 };
+type ProviderError = variant {
+  TooFewCycles : record { expected : nat; received : nat };
+  ServiceUrlParseError : text;
+  ProviderNotFound;
+  ServiceHostNotAllowed : text;
+  NoPermission;
+};
+type ProviderView = record {
+  owner : principal;
+  hostname : text;
+  provider_id : nat64;
+  cycles_per_message_byte : nat64;
+  primary : bool;
+  chain_id : nat64;
+  cycles_per_call : nat64;
+};
+type RegisterProvider = record {
+  credential_headers : opt vec record { text; text };
+  hostname : text;
+  cycles_per_message_byte : nat64;
+  chain_id : nat64;
+  cycles_per_call : nat64;
+  credential_path : text;
+};
+type RejectionCode = variant {
+  NoError;
+  CanisterError;
+  SysTransient;
+  DestinationInvalid;
+  Unknown;
+  SysFatal;
+  CanisterReject;
+};
+type Result = variant { Ok : opt FeeHistory; Err : RpcError };
+type Result_1 = variant { Ok : Block; Err : RpcError };
+type Result_2 = variant { Ok : vec LogEntry; Err : RpcError };
+type Result_3 = variant { Ok : nat; Err : RpcError };
+type Result_4 = variant { Ok : opt TransactionReceipt; Err : RpcError };
+type Result_5 = variant { Ok : SendRawTransactionResult; Err : RpcError };
+type Result_6 = variant { Ok : text; Err : RpcError };
+type Result_7 = variant { Ok : nat; Err : RpcError };
```

Comment:
Is there a good reason not to give these types meaningful names?

Reply:
The Candid file is automatically generated. I brought this up in the weekly Languages team meeting earlier today in case it's possible to improve the ergonomics here.

Reply:
Tracking in #76 (might replace with a Jira ticket depending on what makes the most sense).